Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/DocumentProvider.kt

```diff
@@ -196,13 +246,50 @@ class DocumentProvider : DocumentsProvider() {
 
     private fun findFileNameForNewFile(file: File): File {
```

Comment:
this make duplicate file renamed to "filename (1).jpg" "filename (2).jpg" instead of "filename.1.jpg" "filename.1.2.jpg"